File: STACK_QUEUE/스킬트리.js
function solution(skill, skill_trees) {
  let answer = 0;
  let target_skills = skill_trees.map(tree => tree.split('')
                                .filter(s=>skill.includes(s[0])).join(''));
  target_skills.forEach(target => {
      if (skill.substring(0, target.length)===target){
          answer++;
      }
  })
  return answer;
}

File: STACK_QUEUE/스킬트리.py
def solution(skill, skill_trees):
    answer = 0

    for skill_tree in skill_trees:
        _tree = [s for s in skill_tree if s in skill]
        skill_list = list(skill)
        flag = True
        for t in _tree:
            if skill_list[0] == t:
                skill_list = skill_list[1:]
            else:
                flag = False
        answer = answer+1 if flag == True else answer

    return answer
